rct229/rules/section5/section5rule19.py

```diff
@@ -0,0 +1,132 @@
+from rct229.data_fns.table_G3_111_fns import table_G3_1_1_1_lookup
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_area_type_window_wall_area_dict import (
+    get_area_type_window_wall_area_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.std_comparisons import std_equal
+
+MSG_WARN_MATCHED = "BUILDING IS NOT ALL NEW AND BASELINE WWR MATCHES VALUES PRESCRIBED IN TABLE G3.1.1-1. HOWEVER, THE FENESTRATION AREA PRESCRIBED IN TABLE G3.1.1-1 DOES NOT APPLY TO THE EXISTING ENVELOPE PER TABLE G3.1 BASELINE COLUMN #5 (C). FOR EXISTING ENVELOPE, THE BASELINE FENESTRATION AREA MUST EQUAL THE EXISTING FENESTRATION AREA PRIOR TO THE PROPOSED WORK. A MANUAL CHECK IS REQUIRED TO VERIFY COMPLIANCE."
+MSG_WARN_MISMATCHED = "BUILDING IS NOT ALL NEW AND BASELINE WWR DOES NOT MATCH VALUES PRESCRIBED IN TABLE G3.1.1-1. HOWEVER, THE FENESTRATION AREA PRESCRIBED IN TABLE G3.1.1-1 DOES NOT APPLY TO THE EXISTING ENVELOPE PER TABLE G3.1 BASELINE COLUMN #5(c). FOR EXISTING ENVELOPE, THE BASELINE FENESTRATION AREA MUST EQUAL THE EXISTING FENESTRATION AREA PRIOR TO THE PROPOSED WORK. A MANUAL CHECK IS REQUIRED TO VERIFY COMPLIANCE."
+WWR_THRESHOLD = 0.4
+
+
+class Section5Rule19(RuleDefinitionListIndexedBase):
+    """Rule 19 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule19, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule19.BuildingRule(),
+            index_rmr="baseline",
+            id="5-19",
+            description="For building areas not shown in Table G3.1.1-1, vertical fenestration areas for new buildings and additions shall equal that in the proposed design or 40% of gross above-grade wall area, whichever is smaller.",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule19.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "$": ["building_segments"],
+                    "building_segment": [
```

Comment:
It would be `building_segments`. But I _think_ we don't need lines 45 - 47 at all IF 
- a missing `is_all_new` field is interpreted as False, and 
- a missing `area_type_vertical_fenestration` is interpreted as "OTHER".

These are questions for the RDS team.